ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Block.hs

```diff
@@ -158,14 +173,46 @@ instance
 
 type instance HeaderHash (ShelleyBlock proto era) = ShelleyHash
 
+-- | Reconstruct a Shelley ledger block from a 'ShelleyBlock'.
+--
+-- TODO: we should be able to avoid this conversion in most cases
+fromShelleyBlock :: ShelleyBlock proto era -> SL.Block (ShelleyProtocolHeader proto) era
+fromShelleyBlock blk = SL.Block (shelleyBlockHeader blk) (shelleyBlockBody blk)
+
+-- | Construct a 'ShelleyBlock' from a Shelley ledger block.
+--
+-- TODO: we should be able to avoid this conversion in most cases
+toShelleyBlock ::
+  ShelleyCompatible proto era => SL.Block (ShelleyProtocolHeader proto) era -> ShelleyBlock proto era
+toShelleyBlock (SL.Block hdr body) = mkShelleyBlock hdr body
```

Comment:
We could also just store an `SL.Block`, but encode it differently. But either way, this conversion should be super cheap